docs/collecting-data/collecting-from-own-applications/javascript-trackers/browser-tracker/browser-tracker-v3-reference/advanced-usage/using-an-id-service/index.md

```diff
@@ -0,0 +1,154 @@
+
+This code has minimal functionality and based on our experience can either be:
+
+- An **API endpoint** that is part of the main web application.
+- **A function running in an edge worker.** Almost all modern CDNs allow the addition of ‘middleware’ code which runs on requests at edge nodes in front of the client’s application domain.
+- **A custom middleware based on the customer’s framework**. E.g. ExpressJS middleware, Next.js middleware, Play custom action etc. which can run on every document request.
+- **A low-footprint application with a single endpoint**. A Go web server or something along these lines.
+
+### Developing the ID service code
+
+The responsibilities of this service are:
+1. Create a unique identifier for this browser, set it in a cookie and return it in a `Set-Cookie` header on a domain **accessible** by the Snowplow collector deployed domain. Other cookie header attributes, such as `Domain`, `Secure`, `SameSite`, `httpOnly` etc., should have exactly the same values as the ones set on the collector configuration.
+2. Set the cookie used for the `network_userid` identifier, by default a cookie named `sp` or the value used in `collector.config.name` for the Snowplow collector, to the same value.
+
+_The new unique identifier cookie for sake of simplicity in this document will have the name `spIdService`._
+
+:::tip
+
+Examples showcasing an **accessible** domain in the `Set-Cookie` header for the new identifier:
+
+```js
+/* Collector c.customer.com - Domain customer.com */
+Set-Cookie: spIdService=12345; Domain=customer.com; Secure; HttpOnly
+/* ✅ Works */
+
+/* Collector c.customer.com - Domain customer.com */
+Set-Cookie: spIdService=12345; Domain=c.customer.com; Secure; HttpOnly
+/* ❌ The new update will purge this */
+
+/* Collector c.customer.com - Domain blog.customer.com */
+Set-Cookie: spIdService=12345; Domain=blog.customer.com; Secure; HttpOnly
+/* ❌ Inaccessible to the collector */
+```
+
+:::
+
+### ID service business logic 
+
+The ID service code should include the following logic:
+
+- If the ID service new identifier cookie already exists on the request, then it should re-set the cookies with the same values and updated expiration for both the `spIdService` and `sp` cookies.
+- If `spIdService` does not exist, but the collector's `sp` cookie does exist, then the it should set `spIdService value = sp cookie value`. _This will make sure we keep old identifiers in place and not lose any data._
+- If `spIdService` and `sp` are both missing, then it generates a new ID in the `spIdService` and `sp` cookies with the same unique identifier generation algorithm with the Snowplow pipeline, currently UUID v4.
+
+### Code examples 
+
+Below we showcase a couple of code samples for ID service API endpoints:
+
+<Tabs groupId="id-service" queryString>
+<TabItem value="nextjs" label="Next.js TypeScript" default>
+
+```ts
+export default function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<Data>
```

Comment:
Nit: `NextApiResponse<{ Ok: number }>`

We should document if the response value matters or not, given you've made the examples consistent.

Reply:
Done:

`/* The response needs to return a 200 (OK) status code but any response payload is not necessary. */ ` in the comments

and added a point on the code logic above.